clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -298,12 +312,27 @@
  * Another use case is for a system that maintains local state as described in the previous section. In such a system
  * the consumer will want to initialize its position on start-up to whatever is contained in the local store. Likewise
  * if the local state is destroyed (say because the disk is lost) the state may be recreated on a new machine by
- * reconsuming all the data and recreating the state (assuming that Kafka is retaining sufficient history).
- *
+ * re-consuming all the data and recreating the state (assuming that Kafka is retaining sufficient history).
+ * <p>
  * Kafka allows specifying the position using {@link #seek(TopicPartition, long)} to specify the new position. Special
  * methods for seeking to the earliest and latest offset the server maintains are also available (
  * {@link #seekToBeginning(TopicPartition...)} and {@link #seekToEnd(TopicPartition...)} respectively).
  *
+ * <h4>Prioritizing Partitions for Consumption</h4>
```

Comment:
I wonder if this section would make more sense to average users if phrased in terms of flow control? I see pause() as less about prioritizing partitions and more about implementing back pressure.

(By the way, would it make more sense to have pause and resume accept a collection instead of varargs? Currently pause/resume are the only remaining methods which take varargs)